runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/businessmetrics/BusinessMetricsUtils.kt

```diff
@@ -28,33 +26,49 @@ public val AccountIdBasedEndpointAccountId: AttributeKey<String> = AttributeKey(
 public val ServiceEndpointOverride: AttributeKey<Boolean> = AttributeKey("aws.smithy.kotlin#ServiceEndpointOverride")
 
 /**
- * Emit a business metric to the execution context attributes
+ * Emits a business metric into [MutableAttributes]
+ * @param identifier The identifier of the [BusinessMetric] to be emitted.
  */
 @InternalApi
-public fun ExecutionContext.emitBusinessMetric(metric: BusinessMetric) {
-    if (this.attributes.contains(BusinessMetrics)) {
-        this.attributes[BusinessMetrics].add(metric.identifier)
+public fun MutableAttributes.emitBusinessMetric(identifier: String) {
+    if (this.contains(BusinessMetrics)) {
+        this[BusinessMetrics].add(identifier)
     } else {
-        this.attributes[BusinessMetrics] = mutableSetOf(metric.identifier)
+        this[BusinessMetrics] = mutableSetOf(identifier)
     }
 }
```

Comment:
We need to save the state of credentials business metrics [here](https://github.com/awslabs/aws-sdk-kotlin/blob/02db28ddc186e4eab1dc42edb523dbd9b390a4fc/aws-runtime/aws-config/common/src/aws/sdk/kotlin/runtime/auth/credentials/ProfileCredentialsProvider.kt#L142-L144). Once the metrics are in the business metrics attribute the constants information is gone and all that is left is the identifier.

In hindsight it would've been useful to have the business metrics attribute be a set of business metrics not of set of string and convert them to strings (using their identifiers) before adding them to the UA header.